File: projects/2/michael_ezeigbo/hello.nim
import os

proc main() =
  if paramCount() < 1:
    echo "Usage: ./hello <name> [number]"
    return

  let name = paramStr(1)
  let count = if paramCount() > 1: parseInt(paramStr(2)) else: 1

  if count <= 0:
    echo "Error: The number of greetings must be a positive integer."
    return

  for i in 1..count:
    echo "Hello ", name

main()

File: projects/1/Xavier_Baines/hello.py
import sys

def hello_world(name: str, number: int):
    for _ in range(number):
        print(f"Hello {name}")

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print(f"Usage: python {sys.argv[0]} <name> <number>")
        sys.exit(1)
    
    name = sys.argv[1]
    
    try:
        number = int(sys.argv[2])
        if number <= 0:
            raise ValueError
    except ValueError:
        print("Error: <number> must be a positive integer.")
        sys.exit(1)
    
    hello_world(name, number)

